teuthology/suite/merge.py

```diff
@@ -115,20 +115,22 @@ def config_merge(configs, suite_name=None, **kwargs):
     the entire job (config) from the list.
     """
     seed = kwargs.setdefault('seed', 1)
+    base_config = kwargs.setdefault('base_config', {})
     if not isinstance(seed, int):
         log.debug("no valid seed input: using 1")
         seed = 1
     log.debug("configuring Lua randomseed to %d", seed)
     L.execute(f'local math = require"math"; math.randomseed({seed});')
     new_script = L.eval('new_script')
+    base_config_yaml = yaml.safe_load(str(base_config))
```

Comment:
It's not immediately obvious why we'd need to create a copy of `base_config` here and then again on 133 - could you explain the reasoning?

Reply:
It's to basically enforce a read-only view of the base_config.

Reply:
I tried to do `copy.deepcopy(base_config)` below but ran into inscrutable recursion errors that didn't seem worth debugging. So I'm just reloading it as a legit yaml dictionary here.

Reply:
I think I would suggest just doing a `yaml_complete_obj = copy.deepcopy(base_config.to_dict())` on 133 and dropping this statement.

Reply:
ack

Reply:
done